packages/framework-core/src/booster-command-dispatcher.ts

```diff
@@ -33,17 +38,23 @@ export class BoosterCommandDispatcher {
     const commandClass = commandMetadata.class
     this.logger.debug('Found the following command:', commandClass.name)
 
-    const commandInput = await applyBeforeFunctions(
-      commandEnvelope.value,
-      commandMetadata.before,
-      commandEnvelope.currentUser
-    )
-
-    const commandInstance = createInstance(commandClass, commandInput)
-
-    const register = new Register(commandEnvelope.requestID, commandEnvelope.currentUser, commandEnvelope.context)
-    this.logger.debug('Calling "handle" method on command: ', commandClass)
-    const result = await commandClass.handle(commandInstance, register)
+    let result: unknown
+    let register: Register
+    try {
+      const commandInput = await applyBeforeFunctions(
+        commandEnvelope.value,
+        commandMetadata.before,
+        commandEnvelope.currentUser
+      )
+
+      const commandInstance = createInstance(commandClass, commandInput)
+
+      register = new Register(commandEnvelope.requestID, commandEnvelope.currentUser, commandEnvelope.context)
+      this.logger.debug('Calling "handle" method on command: ', commandClass)
+      result = await commandClass.handle(commandInstance, register)
+    } catch (e) {
+      throw await this.globalErrorDispatcher.dispatch(new CommandHandlerGlobalError(commandEnvelope, e))
```

Comment:
Would it make sense to provide a way to avoid re-throwing the error if it can be handled in the global handler? For instance, we allow the global error dispatcher to return `undefined`, and in this case, we don't throw an error here, but silently ignore the error. If it's different than `undefined` we keep the behavior defined here and re-throw it.

Reply:
It makes sense, in fact, the calls to the dispatch method should be without the `throw` as the dispatch method is the one throwing the exception (I'll fix that):

```typescript
  await this.globalErrorDispatcher.dispatch(new CommandHandlerGlobalError(commandEnvelope, e))
```

The problem with ignoring the error is that it can cause unexpected behaviours. For instances, if there is an error creating the instance in this method, the `register` variable could be undefined and the next call to: `await RegisterHandler.handle(this.config, this.logger, register)` could fail because of that.

Reply:
I'm going to move the register creation outside the try/catch to avoid this problem